README.md

```diff
@@ -24,14 +23,77 @@ If you have downloaded the repo without using the `--recurse-submodules` argumen
 git submodule update --init --recursive
 ```
 
+## coreMQTT Agent Library Configurations
+
+The MQTT Agent library exposes build configuration macros that are required for building the library.
+However, the MQTT Agent library does not include its own configuration file, but exposes configurable macros and their default values at the top of [core_mqtt_agent.h](source/include/core_mqtt_agent.h) and [core_mqtt_agent_command_functions.h](source/include/core_mqtt_agent_command_functions.h). Also, the library inherits any macros defined in coreMQTT's `core_mqtt_config.h`.
+
+To provide custom values for the configuration values, they must be either:
+* Defined in `core_mqtt_config.h` used by coreMQTT
+**OR**
+* Passed as compile time preprocessor macros
+
+## Building the Library
+
+The [mqttAgentFilePaths.cmake](mqttAgentFilePaths.cmake) file contains the information of all source files and the header include path from this repository. In addition to these files, the MQTT Agent library requires the coreMQTT library, whose file information can be obtained from `mqttFilePaths.cmake` in the [coreMQTT submodule](source/dependency/).
+
+For a CMake example of building the MQTT Agent library with the `mqttAgentFilePaths.cmake` file, refer to the `coverity_analysis` library target in [test/CMakeLists.txt](test/CMakeLists.txt) file.
+
+## Building Unit Tests
+
+### Checkout CMock Submodule
+
+To build unit tests, the submodule dependency of CMock is required. Use the following command to clone the submodule:
+```
+git submodule update --checkout --init --recursive test/unit-test/CMock
+```
+
+### Platform Prerequisites
+
+- For running unit tests
+    - **C90 compiler** like gcc
+    - **CMake 3.13.0 or later**
+    - **Ruby 2.0.0 or later** is additionally required for the CMock test framework (that we use).
+- For running the coverage target, **gcov** and **lcov** are additionally required.
+
+### Steps to build **Unit Tests**
+
+1. Go to the root directory of this repository. (Make sure that the **CMock** submodule is cloned as described [above](#checkout-cmock-submodule))
+
+1. Run the *cmake* command: `cmake -S test -B build`
+
+1. Run this command to build the library and unit tests: `make -C build all`
+
+1. The generated test executables will be present in `build/bin/tests` folder.
+
+1. Run `cd build && ctest` to execute all tests and view the test run summary.
+
+## Reference examples
+
+Please refer to the demos of the MQTT Agent library in the following locations for reference examples on FreeRTOS platforms:
+
+| Location |
+| :-: |
+| [coreMQTT Agent Demos](https://github.com/FreeRTOS/coreMQTT-Agent-Demos) |
+| [FreeRTOS/FreeRTOS](https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS-Plus/Demo) |
```

Comment:
Should this specifically point to the Multitask demo?
```suggestion
| [FreeRTOS/FreeRTOS](https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask) |
```

Reply:
Perhaps. I just used the same link that coreMQTT is using, in case the name/directory of the specific demo ever changed

Reply:
I added this change